docs/adr/adr-009-telemetry.md

```diff
@@ -0,0 +1,226 @@
+Basing on the former, the plan is focused on unblocking the reconstruction 
+story first and then proceed with steady covering of our codebase with traces for the complex codepaths as well as 
+metrics and dashboards for "measurables".
+
+Fortunately, the "ShrEx" analysis can be performed with _tracing_ only(more on that in [Tracing](./#Tracing)), so the
+decision for the celestia-node team is to cover with traces only the _necessary_ for the current "ShrEx" stack code as 
+the initial response to the ADR, leaving the rest to be integrated in the background for the devs in the team once they 
+are free as well as for the efficient bootstrapping into the code for the new devs.
+
+The next biggest priority - incentivized Testnet can be largely covered with traces as well. All participant will submit
+traces from their nodes to any provided backend endpoint by us during the whole network lifespan. Later on, we will be 
+able to verify the data of each participant by querying historical traces. This is the feature that some backend solutions 
+provide, which we can use as well to extract valuable insight on how the network performs in macro view.
+
+### Tooling/Dependencies
+
+#### Golang API/Shim
+
+The decision is to use [opentelemetry-go](https://github.com/open-telemetry/opentelemetry-go) for both Metrics and Tracing:
+* Minimal and golang savvy API/shim which gathers years of experience from OpenCensus/OpenMetrics and [CNCF](https://www.cncf.io/)
+* Backends/exporters for all the existing timeseries monitoring DBs, e.g. Prometheus, InfluxDB. As well as tracing backends
+  Jaeger, Uptrace, etc.
+* https://github.com/uptrace/opentelemetry-go-extra/tree/main/otelzap with logging engine with use - Zap
+
+The discussion over this decision can be found in [celestia-node#663](https://github.com/celestiaorg/celestia-node/issues/663)
+and props to @liamsi for initial kickoff and a deep dive into OpenTelemetry.
+
+#### Tracing Backends
+
+For tracing, there are 3 modern OSS tools that are recommended. All of them have bidirectional support with OpenTelemetry:
+* [Uptrace](https://get.uptrace.dev/guide/#what-is-uptrace)
+  * The most recent (~1 year)
+  * The richest UI
+  * Tight to Clickhouse DB
+  * Made by OpenTelemetry
+  * The most lightweight
+* [Jaeger](https://www.jaegertracing.io/)
+  * The most mature
+  * Started by Uber, now supported by CNCF
+  * Supports multiple storages(ScyllaDB, InfluxDB, Amazon DynamoDB)
+* [Graphana Tempo](https://grafana.com/oss/tempo/)
+  * Deep integration with Graphana/Prometheus
+  * Relatively new (~2 years)
+  * Uses Azure, GCS, S3 or local disk for storage
+
+Each of these backends can be used independently and depending on the use case. For us, these are main use cases: 
+* Local development/debugging for the private network or even public network setup
+> I am personally planning to set up the lightweight Uptrace for the local light node. Just to play around and observe 
+> things
```

Comment:
Putting a note after 4 bullet points with what the author personally uses can save the readability. 
Or migrating this info to the `Other` section mentioning the section -> point of the use-case

Reply:
Got you, moved it below. Moving it to the Other section is fat from the context of the quote